modules/renter/persist.go

```diff
@@ -446,3 +477,22 @@ func (r *Renter) LoadSharedFilesASCII(asciiSia string) ([]string, error) {
 	dec := base64.NewDecoder(base64.URLEncoding, bytes.NewBufferString(asciiSia))
 	return r.loadSharedFiles(dec)
 }
+
+// updatePErsistVersionFrom040to133 upgrades a legacy persist file to the next
+// version, adding new fields with their default values.
+func (r *Renter) updatePersistVersionFrom040To133() error {
+	metadata := persist.Metadata{
+		Header:  settingsMetadata.Header,
+		Version: persistVersion040,
+	}
+
+	err := persist.LoadJSON(metadata, &r.persist, filepath.Join(r.persistDir, PersistFilename))
+	if err != nil {
+		return err
+	}
+	metadata.Version = persistVersion133
+	r.persist.MaxDownloadSpeed = DefaultMaxDownloadSpeed
+	r.persist.MaxUploadSpeed = DefaultMaxUploadSpeed
+	r.persist.StreamCacheSize = DefaultStreamCacheSize
+	return persist.SaveJSON(metadata, r.persist, filepath.Join(r.persistDir, PersistFilename))
+}
```

Comment:
this function has some side effects that aren't obvious to the caller, like setting fields in `r.persist`. I suggest instead doing:
```go
func updatePersistVersionFrom040To133(persistPath string) error {
    var p persistence
    LoadJSON(&p, persistPath)
    p.MaxDownloadSpeed = DefaultMaxDownloadSpeed
    // etc.
    return SaveJSON(p, persistPath)
}
```
then it's a bit more clear that the old side effect is rewriting the file.

Reply:
not sure I prefer it that way

Reply:
That requires initializing the map in the persistence, and it also requires the caller to identify the persist path, and it also means that the renter defaults aren't being set directly they must be initialized by reloading the file (which is what we do anyway, but it feels fragile to leave that out of this piece)

Reply:
I don't see the issue with passing the persist path. It makes sense when you think of the function as rewriting a specific file. As for setting defaults, I can see why it might feel fragile to you. "Persist" can refer to both the file on disk and a field in the renter, so it makes sense that when you upgrade one, you should upgrade the other in tandem.

I think this might just be a problem with the name. In other modules, we've called this sort of function `convert`. In the contractor, there's a `convertPersist` function that converts old contractor.json to the new format. Similarly, in the wallet, there's a `convertPersistFrom112To120` function that converts the old wallet.json to a wallet.db. (Both of these functions take filenames as arguments.) "Convert" communicates the intent of the function clearly: it converts one thing into another thing. "Upgrade" is a more ambiguous name, especially when it's a method of the renter instead of a standalone function.

I suggest that we rename the function to `convertPersistFrom040To133` and pass it the filename as an argument. Then it will match the persist conversion functions in the contractor and wallet. Alternatively, we could think of a better name that makes it more obvious that the function also sets fields in the renter.

Reply:
alright